plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -775,6 +775,14 @@ func RunGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercach
 	if !fit {
 		return fit, err
 	}
+	// TODO(HaiyangDING): Make a proper fix for it.
+	allocatable := node.Status.Allocatable
+	allowedPodNumber := allocatable.Pods().Value()
+	if int64(len(nodeInfo.Pods()))+1 > allowedPodNumber {
+		return false,
+			newInsufficientResourceError(podCountResourceName, 1, int64(len(nodeInfo.Pods())), allowedPodNumber)
+	}
+
```

Comment:
It's not that easy. PodFitsResources is a NodeStatus method, which needs predicates.NodeInfo. I can *probably* create one from the things I have here, but it'd be as ugly as this.